Flaky test: Test/StateTransitions_MultipleAddrsEntersReady

```
--- FAIL: Test (18.71s)
    --- FAIL: Test/StateTransitions_MultipleAddrsEntersReady (2.00s)
        tlogger.go:99: clientconn.go:252 [core] parsed scheme: "whatever"  (t=+455.703µs)
        tlogger.go:99: resolver_conn_wrapper.go:100 [core] ccResolverWrapper: sending update to cc: {[{127.0.0.1:37981  <nil> 0 <nil>} {127.0.0.1:37775  <nil> 0 <nil>}] <nil> <nil>}  (t=+551.004µs)
        tlogger.go:99: clientconn.go:1142 [core] Subchannel Connectivity change to CONNECTING  (t=+624.105µs)
        tlogger.go:99: clientconn.go:1253 [core] Subchannel picks a new address "127.0.0.1:37981" to connect  (t=+657.605µs)
        tlogger.go:99: clientconn.go:436 [core] Channel Connectivity change to CONNECTING  (t=+830.806µs)
        tlogger.go:99: clientconn.go:1142 [core] Subchannel Connectivity change to READY  (t=+1.045808ms)
        tlogger.go:99: clientconn.go:436 [core] Channel Connectivity change to READY  (t=+1.070308ms)
        tlogger.go:99: clientconn.go:1142 [core] Subchannel Connectivity change to IDLE  (t=+1.148809ms)
        tlogger.go:99: clientconn.go:436 [core] Channel Connectivity change to IDLE  (t=+1.178109ms)
        clientconn_state_transition_test.go:431: timed out waiting for state 3 (CONNECTING) in flow [CONNECTING READY IDLE CONNECTING]
        tlogger.go:99: clientconn.go:436 [core] Channel Connectivity change to SHUTDOWN  (t=+2.001072763s)
        tlogger.go:99: clientconn.go:1142 [core] Subchannel Connectivity change to SHUTDOWN  (t=+2.001108463s)
FAIL
FAIL	google.golang.org/grpc	36.822s
```

--- a/clientconn_test.go
+++ b/clientconn_test.go
@@ -1122,7 +1122,10 @@ func stayConnected(cc *ClientConn) {
 	ctx, cancel := context.WithTimeout(context.Background(), 10*time.Second)
 	defer cancel()
 
-	for state := cc.GetState(); state != connectivity.Shutdown && cc.WaitForStateChange(ctx, state); state = cc.GetState() {
+	for {
 		cc.Connect()
+		if state := cc.GetState(); state == connectivity.Shutdown || !cc.WaitForStateChange(ctx, state) {
+			return
+		}
 	}
 }

